hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java

```diff
@@ -44,8 +50,20 @@ public class ContractCreateHandler implements TransactionHandler {
      * @throws NullPointerException if one of the arguments is {@code null}
      */
     public TransactionMetadata preHandle(
-            @NonNull final TransactionBody txBody, @NonNull final AccountID payer) {
-        throw new UnsupportedOperationException("Not implemented");
+            @NonNull final TransactionBody txBody,
+            @NonNull final AccountID payer,
+            @NonNull final AccountKeyLookup keyLookup) {
+        final var op = txBody.getContractCreateInstance();
+        final var meta =
+                new SigTransactionMetadataBuilder(keyLookup).txnBody(txBody).payerKeyFor(payer);
+        final var adminKey = asHederaKey(op.getAdminKey());
+        if (adminKey.isPresent() && !((JKey) adminKey.get()).hasContractID()) {
```

Comment:
You do a cast to JKey without checking if it is an instance of

Reply:
All the `HederaKey`s are currently of type `JKey`. This cast will be removed once we remove legacy JKey class